### DISCUSSION TITLE
Alias sub command, make it impossible to use an external completer

### DISCUSSION THREAD

**Discussion by MrPandir** [nushell/nushell#14685] [Q&A] [answered]
# Cause

I'm overriding the `docker ps` commands to make nushell-style output nicer.
Not that it matters, but here's exactly what I'm doing:

```nu
alias "docker ps" = do { ^docker ps --format '{{json .}}' | from json --objects | reject Command CreatedAt Labels LocalVolumes Mounts Networks Ports RunningFor | move Names --before Image | move Size --after Status | table -i false }
```

# Observed behavior

When you type `docker <tab>`, it is automatically expanded to `docker ps`, since this is the only option in nushell, and since the list is not empty **carapace** was not called.

# Desired behavior

When I type `docker <tab>` I expect to see completions from **carapace**.

# Potential solution
One solution I have found that will work is to define all add-ons of the `docker` command in nushell. _But this approach will not take full advantage of **carapace**._

# Question
How can I achieve the expected behavior? If this cannot be done, please let me know.

**Comment by blindFS** [comment, marked as answer]
I believe this issue has been fixed in 0.103 with #15086 .

<img width="694" alt="image" src="https://github.com/user-attachments/assets/d7a88719-3f29-4a99-bd16-1f1cdd79b565" />

**Reply by MrPandir** [reply]
Thanks for notifying me.
